dowhy/causal_estimator.py

```diff
@@ -750,9 +732,21 @@ def estimate_effect(
         logger.error("No valid identified estimand available.")
         return CausalEstimate(None, None, None, control_value=control_value, treatment_value=treatment_value)
 
-    method.update_input(treatment_value, control_value, target_units)
+    if fit_estimator:
+        estimator.fit(
+            data=data,
+            treatment_name=treatment,
+            outcome_name=outcome,
+            effect_modifier_names=effect_modifiers,
+        )
+
+    estimate = estimator.estimate_effect(
```

Comment:
test_significance is not used anywhere in this method. It should be passed to the estimator.

Reply:
`test_significance` is used only on the estimate.add_params, this is because at this point the Estimator is already instantiated by the CausalModel and the `test_significance` was already passed by the `CausalModel.estimate_effect` method, we don't need them as parameter here